File: 3rd_party/adios/python/adios2/operator.py
"""License:
  Distributed under the OSI-approved Apache License, Version 2.0.  See
  accompanying file Copyright.txt for details.
"""


class Operator:
    """High level representation of the Attribute class in the adios2.bindings"""

    def __init__(self, implementation, name):
        self.impl = implementation
        self._name = name

    @property
    def impl(self):
        """Bindings implementation of the class"""
        return self._impl

    @impl.setter
    def impl(self, implementation):
        self._impl = implementation

    def __eq__(self, other):
        if isinstance(other, Operator):
            return self._name == other._name
        return False

    def get_parameters(self):
        """
        Get parameters associated to this Operator

        Returns:
            dict: parameters
        """
        return self.impl.Parameters()

    def set_parameter(self, key, value):
        """
        Set parameter associated to this Operator

        Args:
            str: key
            str: value
        """
        self.impl.SetParameter(key, value)

File: 3rd_party/adios/bindings/CXX11/adios2/cxx11/Operator.cpp
/*
 * Distributed under the OSI-approved Apache License, Version 2.0.  See
 * accompanying file Copyright.txt for details.
 *
 * Operator.cpp :
 *
 *  Created on: Jun 7, 2018
 *      Author: William F Godoy godoywf@ornl.gov
 */

#include "Operator.h"

#include "adios2/core/Operator.h"
#include "adios2/helper/adiosFunctions.h"

namespace adios2
{

Operator::operator bool() const noexcept { return m_Parameters != nullptr; }

std::string Operator::Type() const noexcept { return m_Type; }

void Operator::SetParameter(const std::string key, const std::string value)
{
    helper::CheckForNullptr(m_Parameters, "in call to Operator::SetParameter");
    (*m_Parameters)[key] = value;
}

Params &Operator::Parameters() const
{
    helper::CheckForNullptr(m_Parameters, "in call to Operator::Parameters");
    return *m_Parameters;
}

// PRIVATE
Operator::Operator(const std::string &type, Params *params) : m_Parameters(params), m_Type(type) {}

} // end namespace adios2
